### PR TITLE
feat: fix tools, and lint

### PR DESCRIPTION
**Pull request by arihantbansal**
This PR proposes a couple of changes.

- Fix linting errors
- Create `fetchPrice` function on agent kit

Pls review after #37

### CODE REVIEW

File: src/tools/fetch_price.ts [hunk 1]
```diff
@@ -1,28 +1,21 @@
...
+    const response = await fetch(`https://api.jup.ag/price/v2?ids=${tokenId}`);
 
     if (!response.ok) {
       throw new Error(`Failed to fetch price: ${response.statusText}`);
     }
 
     const data = await response.json();
 
-    const price = data.data[tokenId]?.price;
+    const price = data.data[tokenId.toBase58()]?.price;
```
Comment (thearyanag) [inline src/tools/fetch_price.ts:18]: why is toBase58() here?

Comment (arihantbansal) [reply src/tools/fetch_price.ts:18]: because I changed `tokenId` to be `PublicKey`, inline with the other functions

**Comment by arihantbansal** [comment]
@thearyanag I have again removed the `API References` section from `README.md` which you added back in my other PR. This section is irrelevant now since tools aren't exported, hence users can't call them directly. We can instead specify agent.xyz function reference.

**Comment by arihantbansal** [comment]
After this is merged, making a quick pr for linting and CI (for lint)

**Comment by arihantbansal** [comment]
Also, `pythPriceFeed` function needs work, we can't expect user to provide the price feed ID, need to have function to fetch that given a ticker.

**Comment by arihantbansal** [comment]
> Also, `pythPriceFeed` function needs work, we can't expect user to provide the price feed ID, need to have function to fetch that given a ticker.

I can work on it later.

**Comment by thearyanag** [comment]
> Also, `pythPriceFeed` function needs work, we can't expect user to provide the price feed ID, need to have function to fetch that given a ticker.

agreed https://github.com/sendaifun/solana-agent-kit/issues/19#issuecomment-2558958617

**Pull request merged**